src/m365/yammer/commands/yammer-search.ts

```diff
@@ -0,0 +1,214 @@
+import { Logger } from '../../../cli';
+import { CommandOption } from '../../../Command';
+import GlobalOptions from '../../../GlobalOptions';
+import request from '../../../request';
+import YammerCommand from '../../base/YammerCommand';
+import commands from '../commands';
+
+interface CommandArgs {
+  options: Options;
+}
+
+interface Options extends GlobalOptions {
+  search: string;
+  limit?: number;
+}
+
+class YammerSearchCommand extends YammerCommand {
+  private summary: any;
+  private messages: any[];
+  private groups: any[];
+  private topics: any[];
+  private users: any[];
+
+  constructor() {
+    super();
+    this.summary = {
+      messages: 0,
+      groups: 0,
+      topics: 0,
+      users: 0
+    }
+    this.messages = [];
+    this.groups = [];
+    this.topics = [];
+    this.users = [];
+  }
+
+  public get name(): string {
+    return `${commands.YAMMER_SEARCH}`;
+  }
+
+  public get description(): string {
+    return 'Returns a list of messages, users, topics and groups that match the user’s search query.';
+  }
+
+  public getTelemetryProperties(args: CommandArgs): any {
+    const telemetryProps: any = super.getTelemetryProperties(args);
+    telemetryProps.search = typeof args.options.search !== 'undefined';
+    telemetryProps.limit =  typeof args.options.limit !== 'undefined';
+    return telemetryProps;
+  }
+
+  private getAllItems(logger: Logger, args: CommandArgs, page: number): Promise<void> {
+    return new Promise<void>((resolve: () => void, reject: (error: any) => void): void => {
+      let endpoint = `${this.resource}/v1`;
+      endpoint += `/search.json?search=${args.options.search}&page=${page}`;
+
+      const requestOptions: any = {
+        url: endpoint,
+        headers: {
+          accept: 'application/json;odata.metadata=none',
+          'content-type': 'application/json;odata=nometadata'
+        },
+        responseType: 'json'
+      };
+
+      request
+        .get(requestOptions)
+        .then((results: any): void => {
```

Comment:
Since we're using properties of this object further in code, let's define its shape and not use `any` to avoid runtime errors.

```suggestion
        .then((results: YammerSearchResponse): void => {
```